packages/devtools/lib/src/memory/memory.dart

```diff
@@ -307,6 +314,10 @@ class MemoryScreen extends Screen with SetStateMixin {
     table.onSelect.listen((ClassHeapDetailStats row) async {
       ga.select(ga.memory, ga.inspectClass);
 
+      if (tableContainer.element.children.length > 2) {
```

Comment:
nit: perhaps store a reference to the specific child you are wanting to remove or comment which child it is for clarity?

Reply:
That was an odd conditional since only, at most, three children can exist and its the third that is the issue.

```
      // User selected a new instance from the list of class instances so the
      // instance view which would be the third child needs to be removed.
      if (tableContainer.element.children.length == 3) {

```